File: benchmarks/python/format_interp.py
def format_interp_bench(iterations: int) -> int:
    sink = 0
    i = 0
    while i < iterations:
        s = f"v={i}"
        sink += len(s) + ord(s[-1])
        i += 1
    return sink


def main() -> int:
    iterations = 1000
    result = format_interp_bench(iterations)
    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: benchmarks/rust/format_interp.rs
fn format_interp_bench(iterations: i64) -> i64 {
    let mut sink: i64 = 0;
    let mut i: i64 = 0;
    while i < iterations {
        let s = format!("v={}", i);
        sink += s.len() as i64 + s.as_bytes()[s.len() - 1] as i64;
        i += 1;
    }
    sink
}

fn main() {
    let iterations: i64 = 1_000;
    let result = format_interp_bench(iterations);
    println!("{}", result);
}
